lib/galaxy/jobs/runners/util/job_script/__init__.py

```diff
@@ -124,18 +126,23 @@ def _handle_script_integrity(path, config):
                 script_integrity_verified = True
                 break
 
+            log.debug("Script integrity error: returncode was %d", returncode)
+
             # Else we will sync and wait to see if the script becomes
             # executable.
             try:
                 # sync file system to avoid "Text file busy" problems.
                 # These have occurred both in Docker containers and on EC2 clusters
                 # under high load.
                 subprocess.check_call(INTEGRITY_SYNC_COMMAND)
-            except Exception:
-                pass
+            except Exception as e:
+                log.debug("Error syncing the filesystem %s", unicodify(e))
+
             time.sleep(sleep_amt)
-        except Exception:
-            pass
+        except OSError as ose:
+            log.debug("Script not available yet %s", unicodify(ose))
```

Comment:
These 2 lines are redundant since the error message is the same as below and `OSError` is a subclass of `Exception`.

Reply:
Originally I'd had a different behaviour here. Will collapse.